ethereum/block/wait_for_receipt.go

```diff
@@ -0,0 +1,51 @@
+package block
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	// in total, we will wait up to 1 hour for a transaction to be included
+	maxRetries           = 360
+	receiptSleepDuration = time.Second * 10
+)
+
+var (
+	log               = logging.Logger("ethereum/block")
+	errReceiptTimeOut = errors.New("failed to get receipt, timed out")
+)
+
+// WaitForReceipt waits for the transaction to be mined into a block. If the transaction was reverted when mined,
+// we return an error describing why.
```

Comment:
It actually waits until the transaction has been mined into a block plus some number of confirmations. The documentation for `TransactionReceipt` is not clear on what that number of confirmations is. Empirically, it appears to be 9 confirmations. If I get the current block after waiting for the receipt, it is 9 blocks ahead of the value in `receipt.BlockNumber`.